astropy/modeling/tests/test_astro_sherpa.py

```diff
@@ -0,0 +1,242 @@
+
+        self.model2d = Gaussian2D(amplitude=10, x_mean=5, y_mean=6, x_stddev=0.8, y_stddev=1.1)
+        self.model2d.theta.fixed = True
+
+        self.yy = self.model2d(self.xx1, self.xx2)
+        self.dxx1 = err * np.random.uniform(0.5, 1., size=self.xx1.size)
+        self.dxx2 = err * np.random.uniform(0.5, 1., size=self.xx2.size)
+        self.dyy = err * np.random.uniform(0.5, 1., size=self.yy.size)
+
+        self.tmodel2d = self.model2d.copy()
+        self.model2d.amplitude = 5
+        self.model2d.x_mean = 6
+        self.model2d.y_mean = 5
+        self.model2d.x_stddev = 0.2
+        self.model2d.y_stddev = 0.7
+
+        # to stop stddev going negitive and getting div by zero error
+        self.model1d.stddev.min = 1e-99
+        self.model1d_2.stddev.min = 1e-99
+        self.model2d.x_stddev.min = 1e-99
+        self.model2d.y_stddev.min = 1e-99
+
+        self.fitter = SherpaFitter(statistic=Chi2())
+
+    def test_make_datasets_single(self):
+        data_noerr = make_datasets(1, x=self.x1, y=self.y1)
+        assert_allclose(data_noerr.get_x(), self.x1)
+        assert_allclose(data_noerr.get_y(), self.y1)
+
+        data_xerr = make_datasets(1, x=self.x1, y=self.y1, xerr=self.dx1)
+        assert_allclose(data_xerr.get_x(), self.x1)
+        assert_allclose(data_xerr.get_y(), self.y1)
+        assert_allclose(data_xerr.get_xerr(), 2 * self.dx1)
+
+        data_yerr = make_datasets(1, x=self.x1, y=self.y1, yerr=self.dy1)
+        assert_allclose(data_yerr.get_x(), self.x1)
+        assert_allclose(data_yerr.get_y(), self.y1)
+        assert_allclose(data_yerr.get_yerr(), self.dy1)
+
+        data_botherr = make_datasets(1, x=self.x1, y=self.y1, yerr=self.dy1, xerr=self.dx1)
+        assert_allclose(data_botherr.get_x(), self.x1)
+        assert_allclose(data_botherr.get_y(), self.y1)
+        assert_allclose(data_botherr.get_yerr(), self.dy1)
+        assert_allclose(data_botherr.get_xerr(), 2 * self.dx1)
+
+    def test_make_datasets_2d(self):
+        data_noerr = make_datasets(2, x=self.xx1, y=self.xx2, z=self.yy)
+        assert_allclose(data_noerr.get_x0(), self.xx1)
+        assert_allclose(data_noerr.get_x1(), self.xx2)
+        assert_allclose(data_noerr.get_y(), self.yy)
+
+        data_xyerr = make_datasets(2, x=self.xx1, y=self.xx2, xerr=self.dxx1, yerr=self.dxx2, z=self.yy)
+        assert_allclose(data_xyerr.get_indep(), np.vstack([self.xx1 - self.dxx1, self.xx2 - self.dxx2, self.xx1 + self.dxx1, self.xx2 + self.dxx2]))
+        assert_allclose(data_xyerr.get_y(), self.yy)
+
+        data_xyzerr = make_datasets(2, x=self.xx1, y=self.xx2, xerr=self.dxx1, yerr=self.dxx2, z=self.yy, zerr=self.dyy)
+        assert_allclose(data_xyerr.get_indep(), np.vstack([self.xx1 - self.dxx1, self.xx2 - self.dxx2, self.xx1 + self.dxx1, self.xx2 + self.dxx2]))
+        assert_allclose(data_xyzerr.get_y(), self.yy)
+        assert_allclose(data_xyzerr.get_yerr(), self.dyy)
+
+    def test_make_datasets_multi(self):
```

Comment:
In this case for example, I had to go back and look at the code you are testing to see what exactly the "multi" is. Again 1-2 sentences of text will make it a lot faster for someone who comes in 3 years form now to understand this code.